src/argparcel/docstrings.py

```diff
@@ -0,0 +1,59 @@
+"""Utility to parse 'field docstrings' from a dataclass.
+
+Static-analysis tools like pyright will interpret strings that appear immediately after
+the definition of a field in a dataclass as documentation for that field, so this is an
+'expected' concept. However, it is _not_ a feature of the Python runtime.
+
+For that reason we resort to some dastardly AST inspection to extract the messages.
+"""
+
+from __future__ import annotations
+
+import ast
+import inspect
+import itertools
+import typing
+
+if typing.TYPE_CHECKING:
+    import _typeshed
+
+
+def get_field_docstrings(cls: type[_typeshed.DataclassInstance]) -> dict[str, str]:
+    """Extracts any 'field docstrings' present in `cls`.
+
+    Returns a mapping from field name -> docstring, only for fields with which strings
+    can be associated.
+    """
+    # Obtaining the source code might fail with an exception; we let that propagate up
+    # to the user.
+    source = inspect.getsource(cls)
+    tree = ast.parse(source)
+
+    # Find the class definition node; if it doesn't exist something is fundamentally
+    # broken, so we are happy with an assert.
```

Comment:
What if the source text changed between startup and the execution of L29? I have seen that confuse tracebacks in the past. I think we might be able to provide a useful error message here.

I'm happy to issueise and try to trigger the assert when I have some time, if you like.

Reply:
#14